lib/orocos/process.rb

```diff
@@ -1171,9 +1199,25 @@ def self.wait_running(process, timeout = nil, name_service = Orocos::CORBA.name_
                     block.call(process)
                 else
                     # Get any task name from that specific deployment, and check we
-                    # can access it. If there is none
+                    # can access it either through the ior pipe or through the CORBA name
+                    # service.
+                    if ior_read?
+                        ior_message += ior_read.read_nonblock 1
+                        task_name, ior, msg_status = parse_ior_message ior_message
```

Comment:
`task_name` refers to which task?

Reply:
It would be from the one we get from the message